src/OpenLoco/src/Vehicles/VehicleHead.cpp

```diff
@@ -179,12 +182,158 @@ namespace OpenLoco::Vehicles
         calculateRefundCost();
     }
 
+    // 0x004B8340
+    static void RecalculateTrainMinReliability(VehicleHead& head)
+    {
+        Vehicle train(head);
+        const auto getReliabilty = [](const Car& car) -> uint16_t {
+            if (car.front->reliability == 0)
+            {
+                return 0xFFFFU;
+            }
+            return car.front->reliability;
+        };
+        const auto minReliability = (*std::min_element(train.cars.begin(), train.cars.end(), [&getReliabilty](const Car& carLhs, const Car& carRhs) {
+                                        return getReliabilty(carLhs) < getReliabilty(carRhs);
+                                    })).front->reliability;
+        train.veh2->reliability = minReliability == 0xFFFFU ? 0 : minReliability / 256;
+    }
+
     // 0x004B9509
     void VehicleHead::updateDaily()
     {
-        registers regs{};
-        regs.esi = X86Pointer(this);
-        call(0x004B9509, regs);
+        bool resetStoppedTimeout = true;
+        if (status == Status::stopped)
+        {
+            if (mode == TransportMode::road)
+            {
+                if (tileX != -1
+                    && hasVehicleFlags(VehicleFlags::commandStop))
+                {
+                    if (Tutorial::state() == Tutorial::State::none)
+                    {
+                        var_79++;
+                        if (var_79 >= 20)
+                        {
+                            GameCommands::VehicleChangeRunningModeArgs args{};
+                            args.head = head;
+                            args.mode = GameCommands::VehicleChangeRunningModeArgs::Mode::startVehicle;
+                            auto regs = static_cast<Interop::registers>(args);
+                            regs.bl = GameCommands::Flags::apply;
+                            GameCommands::vehicleChangeRunningMode(regs);
+                        }
+                        else
+                        {
+                            resetStoppedTimeout = false;
+                        }
+                    }
+                    else
+                    {
+                        resetStoppedTimeout = false;
+                    }
+                }
+            }
+            else
+            {
+                resetStoppedTimeout = false;
+            }
+        }
+        if (resetStoppedTimeout)
+        {
+            var_79 = 0;
+        }
+
+        if (status == Status::crashed || status == Status::stuck)
+        {
+            crashedTimeout = Math::Bound::add(crashedTimeout, 1U);
+            if (!CompanyManager::isPlayerCompany(owner))
+            {
+                if (crashedTimeout > 14 && var_60 != 0xFF)
```

Comment:
can we add a constant for the `14`, and can we rename `var_60`?

Reply:
I've added a constant but again don't know enough about var_60 atm.